### PR TITLE
implement output-field method

### PR DESCRIPTION
**Pull request by millay1114**
implement create field method.
please review.

### CODE REVIEW

File: app/game/field.py [hunk 1]
```diff
@@ -156,17 +169,36 @@ def check_obstacle(self, point):
             return False
         return True
 
+    def get_field_object(self, point=None, x=None, y=None):
+        x_point = point.x if point is not None else x
+        y_point = point.y if point is not None else y
+
+        if self.bombs[x_point][y_point] is not None:
```
Comment (kouki-dan) [inline app/game/field.py]: Use field_objet function

File: app/game/field.py [hunk 2]
```diff
@@ -156,17 +169,36 @@ def check_obstacle(self, point):
...
+            return self.bombs[x_point][y_point]
+
+        if self.items[x_point][y_point] is not None:
+            return self.bombs[x_point][y_point]
```
Comment (kouki-dan) [inline app/game/field.py]: Is it mistake?
You mean `self.items`?

File: app/game/field_outputter.py [hunk 3]
```diff
@@ -1,24 +1,70 @@
...
 
 class FieldOutputter:
     recent_field_ts = {}
 
     @classmethod
     def post_field(cls, channel, field, new_message=False):
-        field_text = "TODO: create this by field"
+
+        field_text = ""
+        for y in range(0, field.y_size):
```
Comment (kouki-dan) [inline app/game/field_outputter.py]: A first argument of range function can eliminate.
If range function called with a single argument, it means 0 to {first_argument} iterator

File: app/game/bomber.py [hunk 4]
```diff
@@ -41,22 +43,28 @@ def __init__(self, channel, users):
 
     def start(self):
         self.running = True
+        FieldOutputter.post_field(self.channel, self.field)
```
Comment (kouki-dan) [inline app/game/bomber.py]: test

Comment (kouki-dan) [inline app/game/bomber.py]: okay?

Comment (kouki-dan) [inline app/game/bomber.py]: why?

File: app/game/field.py (hunk 2 above)
Comment (millay1114) [inline app/game/field.py]: oops

File: app/game/field.py (hunk 1 above)
Comment (millay1114) [inline app/game/field.py]: ok

**Comment by millay1114** [comment]
fix it.

File: app/game/field.py [hunk 5]
```diff
@@ -216,17 +220,39 @@ def check_obstacle(self, point):
...
+            return Object.block
+
+        if field_object(self.bombs, point) is not None:
+            return self.bombs[point.x][point.y]
+
+        if field_object(self.items, point) is not None:
+            return self.items[point.x][point.y]
+
+        for person in self.persons:
+            if person.point.x is point.x and person.point.y is point.y:
```
Comment (kouki-dan) [inline app/game/field.py]: is judge identity of object. 
In python, small int object is reusable so 1 is 1 and 1 is (1+1-1) are also True, but big number is not so.
For example, 12345678901234567890 is (12345678901234567890+1-1) is False.
You can use == for judging equality for variable.

File: app/game/field_outputter.py [hunk 6]
```diff
@@ -1,24 +1,74 @@
...
         if new_message or channel not in cls.recent_field_ts:
             res = slacker.chat.post_message(channel, field_text)
             cls.recent_field_ts[channel] = res.body["ts"]
             cls.add_controller(channel, res.body["ts"])
         else:
             ts = cls.recent_field_ts[channel]
             import random
-            slacker.chat.update(channel, ts, field_text+"(edited)"+str(random.random())) # TODO: remove this string
+            # TODO: remove this string
+            slacker.chat.update(
```
Comment (kouki-dan) [inline app/game/field_outputter.py]: The TODO should remove in this PR. and below str(random.random()) should also remove

File: app/game/field.py [hunk 7]
```diff
@@ -235,18 +261,44 @@ def is_out_of_source(source, point):
...
+
         return {
-            "objects":  [ [Object.empty]*y_size for x in range(x_size)],
-            "items": [ [None]*y_size for x in range(x_size)],
+            "objects":  objects,
+            "items": items,
             "person_initial_positions": [
-                Point(0, 0),
-                Point(x_size-1, y_size-1),
+                Point(1, 1),
```
Comment (kouki-dan) [inline app/game/field.py]: use person num for this

Comment (millay1114) [inline app/game/field.py]: How do i use person num?

Comment (kouki-dan) [inline app/game/field.py]: your code makes a position for 4 persons regardless of person_num

File: app/game/field_outputter.py (hunk 6 above)
Comment (millay1114) [inline app/game/field_outputter.py]: i see.

File: app/game/field.py (hunk 7 above)
Comment (millay1114) [inline app/game/field.py]: OK

**Comment by millay1114** [comment]
please review me

File: app/game/field.py [hunk 8]
```diff
@@ -111,20 +113,21 @@ def add_fire(self, count=1):
...
-        self.bombs = [ [None]*y_size for x in range(x_size)]
+        self.bombs = [[None] * y_size for x in range(x_size)]
         map_ = FieldCreater.generate_map(x_size, y_size, len(users))
         self.objects = map_["objects"]
         self.items = map_["items"]
-        self.persons = [ Person(user, initial_pos)
-              for user, initial_pos in zip(users, map_["person_initial_positions"])
-            ]
+        self.persons = [Person(user, initial_pos, num)
+                        for user, initial_pos, num in zip(users, map_["person_initial_positions"], range(0, len(users)))]
```
Comment (kouki-dan) [inline app/game/field.py]: range(0

File: app/game/field.py [hunk 9]
```diff
@@ -165,7 +169,7 @@ def fire_bomb(self, point):
                     put_object_to_field(self.bombs, fire_pos, FiredPerson(person))
             object = field_object(self.objects, fire_pos)
             if object == Object.block:
-                put_object_to_field(self.objects, fire_pos, None)
+                put_object_to_field(self.objects, fire_pos, Object.empty)
```
Comment (kouki-dan) [inline app/game/field.py]: sorry

File: app/game/field.py [hunk 10]
```diff
@@ -216,17 +220,39 @@ def check_obstacle(self, point):
...
+
+        if field_object(self.objects, point) is Object.block:
+            return Object.block
+
+        if field_object(self.bombs, point) is not None:
+            return self.bombs[point.x][point.y]
+
+        if field_object(self.items, point) is not None:
+            return self.items[point.x][point.y]
+
```
Comment (kouki-dan) [inline app/game/field.py]: Why don't use a return value of field_object function?

File: app/game/field.py [hunk 11]
```diff
@@ -235,18 +261,48 @@ def is_out_of_source(source, point):
...
         return {
-            "objects":  [ [Object.empty]*y_size for x in range(x_size)],
-            "items": [ [None]*y_size for x in range(x_size)],
+            "objects":  objects,
+            "items": items,
             "person_initial_positions": [
-                Point(0, 0),
-                Point(x_size-1, y_size-1),
+                person_initial_positions[num] for num in range(person_num)
```
Comment (kouki-dan) [inline app/game/field.py]: you mean person_initial_position[:num]？

**Comment by kouki-dan** [comment]
commented

File: app/game/field.py (hunk 10 above)
Comment (millay1114) [inline app/game/field.py]: because it does not need to call ``is_out_of_source`` function.

File: app/game/field.py (hunk 8 above)
Comment (millay1114) [inline app/game/field.py]: ;-)

File: app/game/field.py (hunk 10 above)
Comment (millay1114) [inline app/game/field.py]: sorry, i was misunderstanding

**Comment by millay1114** [comment]
fixed.

**Pull request merged**